actionpack/lib/action_dispatch/journey/routes.rb

```diff
@@ -57,7 +57,7 @@ def ast
 
       def simulator
         @simulator ||= begin
-          gtg = GTG::Builder.new(ast).transition_table
+          gtg = GTG::Builder.new(ast).transition_table unless ast.blank?
```

Comment:
Is this not fixing the symptoms instead of the cause? Why the ast is blank? It should never be.

Reply:
The '#ast' method returns `nil` when there are no anchored routes (notice there is an `unless` condition there). I assumed that "not having anchored routes" is a valid condition, so my patch was handling that case too. 

```ruby
      def ast
        @ast ||= begin
          asts = anchored_routes.map(&:ast)
          Nodes::Or.new(asts) unless asts.empty?
        end
      end
```

So should there always be at least one anchored route?

Reply:
Oh. Now I see. I was just asking because the fix seemed fixing a symptom. But should the simulator be built even without a gtg? I don't think so because it will also fail if there is no a transition table.

Reply:
So [`#simulator` is invoked when eager loading the `Router`](https://github.com/jorgemanrubia/rails/blob/c40b4428e6d3885a8adc0ceba3aeac7599c14879/actionpack/lib/action_dispatch/journey/router.rb#L30). Maybe a better patch would be so skip that call when `ast` is blank?

Reply:
I think you patch is correct, but we should return `nil` when `simulator` is called and `ast` is not present.

Reply:
Other thing to investigate. Why RouteSet is being inserted automatically by ActiveStorage?